src/addrdb.cpp

```diff
@@ -119,18 +175,46 @@ bool DeserializeFileDB(const fs::path& path, Data& data)
 }
 } // namespace
 
-CBanDB::CBanDB(fs::path ban_list_path) : m_ban_list_path(std::move(ban_list_path))
+CBanDB::CBanDB(fs::path ban_list_path)
+    : m_banlist_dat(ban_list_path.string() + ".dat"),
+      m_banlist_json(ban_list_path.string() + ".json")
 {
 }
 
 bool CBanDB::Write(const banmap_t& banSet)
 {
-    return SerializeFileDB("banlist", m_ban_list_path, banSet);
+    std::vector<std::string> errors;
+    if (util::WriteSettings(m_banlist_json, {{JSON_KEY, BanMapToJson(banSet)}}, errors)) {
+        return true;
+    }
+
+    for (const auto& err : errors) {
+        error("%s: %s", __func__, err);
```

Comment:
same commit:

* It seems the caller ignores the return code, so calling this ERROR seems confusing
* No longer need `__func__`

Reply:
The caller of `CBanDB::Write()` does not ignore its return code? If you mean the return code of `error()` itself - this is because we want to log all of the errors (possibly more than one) before returning `false`.

Removed `__func__`.